spec/models/manageiq/providers/ibm_power_hmc/infra_manager/host_spec.rb

```diff
@@ -0,0 +1,35 @@
+describe ManageIQ::Providers::IbmPowerHmc::InfraManager::Host do
+  let(:ems) do
+    FactoryBot.create(:ems_ibm_power_hmc_infra_with_authentication)
+  end
+
+  let(:host) do
+    FactoryBot.create(:ibm_power_hmc_host, :ext_management_system => ems, :ems_ref => "d47a585d-eaa8-3a54-b4dc-93346276ea37")
+  end
+
+  let(:samples) do
+    JSON.parse(Pathname.new(__dir__).join(filename).read)["test_data"]
+  end
+
+  context "host" do
+    let(:filename) { "test_data/metrics_host.json" }
+    it "process_samples" do
+      expect(host.process_samples(ManageIQ::Providers::IbmPowerHmc::InfraManager::MetricsCapture::VIM_STYLE_COUNTERS, samples)).to include(
```

Comment:
Could you `allow(host).to receive(:collect_samples).and_return(samples)` then just call `host.perf_collect_metrics("realtime")` ?  Best to test the top-level method if at all possible

Reply:
Ok, nice Rspec tip